persistent/Database/Persist.hs

```diff
@@ -295,6 +298,74 @@ f >=. a  = Filter f (Left a) Ge
 infix 4 <-., /<-.
 (<-.), (/<-.) :: forall v typ.  PersistField typ => EntityField v typ -> [typ] -> Filter v
 
+
+-- |
+-- Check if value is like given pattern. This function takes
+-- field, a character to escape wildcards and a like pattern
+--
+-- @since 2.9.2
+--
+-- === __Example usage__
+--
+-- @
+-- selectLikeSimon :: MonadIO m => ReaderT SqlBackend m ()
+-- selectLikeSimon = selectList [likeWithEscape '@' UserName "%Si%"]
+-- @
+--
+-- The above query when applied on dataset-1, will produce this:
+--
+-- > +-----+-----+-----+
+-- > |id   |name |age  |
+-- > +-----+-----+-----+
+-- > |2    |Simon|41   |
+-- > +-----+-----+-----+
+--
+-- and the SQL like the following:
+--
+-- > SELECT id, name, age, color FROM Person WHERE name LIKE "%si%" '@'
```

Comment:
Nitpick: The function has `"%Si%"` but this example has `%si%`. Is the function case sensitive?